doc/source/whatsnew/v2.1.2.rst

```diff
@@ -39,7 +39,7 @@ Bug fixes
 Other
 ~~~~~
 - Fixed non-working installation of optional dependency group ``output_formatting``. Replacing underscore ``_`` with a dash ``-`` fixes broken dependency resolution. A correct way to use now is ``pip install pandas[output-formatting]``.
--
+- Setting the environment variable ``PANDAS_INFER_STRING`` to ``"1"`` will now enable the option ``pd.options.future.infer_string`` (:issue:`55533`)^
```

Comment:
why?  i think we're trying to move away from the environment variable pattern.

also typo trailing '^'

Reply:
This is just easier to opt in compared with the option.

I thought we are trying to move away from global options independently ov env variables.

Removed the typo.

Reply:
need this for the ci build as well

Reply:
for the CI could we have the env option get checked at the top of conftest?

Reply:
Can you clarify that comment?

Reply:
instead of changing config_init (which affects users), at the top of pandas/conftest.py put

```
if os.environ.get("PANDAS_INFER_STRING", False):
    config.set_option(...)
```

Then we can set the env variable for the relevant CI build without making this a user-facing option.